### ISSUE TITLE
[🐞] `jsx-key` lint rule doesn't work as expected

### ISSUE THREAD

**Issue by ahnpnl** [BuilderIO/qwik#3366]
### Which component is affected?

Starters / CLI

### Describe the bug

In `eslint-plugin-qwik`, there is a rule `jsx-key` which is set to `warn` for `recommended` config and `error` for `strict` config, see https://github.com/BuilderIO/qwik/blob/c84f82199b60c9fe5640fd36d5d89526332413b2/packages/eslint-plugin-qwik/index.ts#L29 but it doesn't work as expected with newly created Qwik project.

### Reproduction

https://github.com/ahnpnl/my-qwik-app

### Steps to reproduce

- Check out the repo
- Install deps with `pnpm`
- Open `src/routes/todolist/index.tsx`, there is a piece of codes which uses `Array.prototype.map` to render `<li>`. These `<li>` doesn't have `key` attribute
- Run `pnpm run lint` and see no warnings in console

### System Info

```shell
System:
    OS: macOS 13.2.1
    CPU: (8) arm64 Apple M1
    Memory: 75.91 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.1 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.3 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 111.0.5563.64
    Firefox: 108.0.2
    Safari: 16.3
  npmPackages:
    @builder.io/qwik: 0.21.0 => 0.21.0 
    @builder.io/qwik-city: 0.5.3 => 0.5.3 
    undici: 5.20.0 => 5.20.0 
    vite: 4.1.4 => 4.1.4
```

### Additional Information

**N.A.**

**Comment by manucorporat** [comment]
We havent released this new rules yet!

**Comment by ahnpnl** [comment]
ah crab! Look like it's included after **0.22.0**?

**Comment by ahnpnl** [comment]
thanks it works with **0.22.0**
